routes/volunteers_routes.js

```diff
@@ -0,0 +1,209 @@
+var volunteers_model = require('../models/volunteers');
+var Department = volunteers_model.Department;
+var Role = volunteers_model.VolunteerRole;
+var Volunteer = volunteers_model.Volunteer;
+//const userRole = require('../libs/user_role');
+var DrupalAccess = require('../libs/drupal_acces').DrupalAccess;
+var log = require('../libs/logger')(module);
+
+//roles ... consider moving to some other file.
+const VOLUNTEER_MANAGER = 0;
+const VOLUNTEER_DEPT_MANAGER = 2;
+const CURRENT_EVENT = 0;
+//GET /volunteer/departments
+var get_departments = function (req, res) {
+    Department.fetchAll().then(function (deps) {
+        res.json(deps.toJSON());
+    }).catch((err) => {
+        res.status(500).json({
+            error: true,
+            data: {
+                message: err.message
+            }
+        });
+    });
+};
+//GET /volunteer/volunteers
+var get_roles = function (req, res) {
+    Role.fetchAll().then(function (deps) {
+        res.json(deps.toJSON());
+    }).catch((err) => {
+        res.status(500).json({
+            error: true,
+            data: {
+                message: err.message
+            }
+        });
+    });
+}
+
+function __has_permissions(user_id, perm_level) {
+    return new Promise((resolve, reject) => {
+        //TODO - check if loggedin user has permissions...
+        resolve();
+
+    });
+};
+
+function __merge_volunteer_info(vol_data_model, user_info) {
+    return {
```

Comment:
this whole validation area: `user_info ? ...` can be replaced with `_.get(user_info, 'first_name', undefined) `